### ISSUE TITLE
gleap_dio_interceptor for Flutter < v#

### ISSUE THREAD

**Issue by 1MochaChan1** [GleapSDK/Flutter-SDK#1]
`flutter doctor`
`Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 2.10.4, on Microsoft Windows [Version 10.0.19044.1766], locale en-IN)
[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.1.4)
[√] Android Studio (version 2020.3)
[√] VS Code (version 1.68.1)
[√] Connected device (4 available)
[√] HTTP Host Availability`

Currently I am importing the `gleap_sdk` package as such:
```
gleap_sdk:
    git:
      url: https://github.com/GleapSDK/Flutter-SDK.git
      ref: flutter-v2
```

Now when I try to add `gleap_dio_interceptor` it says :
```
[discoveri-flutter-app] flutter pub add gleap_dio_interceptor
Because gleap_dio_interceptor <1.0.1 depends on gleap_sdk ^6.2.2 and gleap_dio_interceptor >=1.0.1 <1.0.2 depends on gleap_sdk ^6.2.3, gleap_dio_interceptor <1.0.2 requires gleap_sdk ^6.2.2.
And because gleap_dio_interceptor >=1.0.2 <1.0.3 depends on gleap_sdk ^6.3.0 and gleap_dio_interceptor >=1.0.3 <1.0.5 depends on gleap_sdk ^6.3.1, gleap_dio_interceptor <1.0.5 requires gleap_sdk ^6.2.2.
And because gleap_dio_interceptor >=1.0.5 <1.0.6 depends on gleap_sdk ^6.3.2 and gleap_dio_interceptor >=1.0.6 <1.0.8 depends on gleap_sdk ^6.3.3, gleap_dio_interceptor <1.0.8 requires gleap_sdk ^6.2.2.
And because gleap_dio_interceptor >=1.0.8 <1.0.9 depends on gleap_sdk ^6.3.5 and gleap_dio_interceptor >=1.0.9 <1.1.0 depends on gleap_sdk ^6.3.6, gleap_dio_interceptor <1.1.0 requires gleap_sdk ^6.2.2.
And because gleap_dio_interceptor >=1.1.0 <1.2.0 depends on gleap_sdk >=6.3.7 and gleap_dio_interceptor >=1.2.0 depends on gleap_sdk >=6.4.4, every version of gleap_dio_interceptor requires gleap_sdk from hosted.
So, because discoveri depends on both gleap_sdk from git and gleap_dio_interceptor any, version solving failed.
pub finished with exit code 65
exit code 65
```
In summary it says:

> every version of gleap_dio_interceptor requires gleap_sdk from hosted

Can someone suggest a workaround to this? (Except upgrading my Flutter to v3)

**Comment by 1MochaChan1** [comment]
For now I've upgraded to Flutter v3, so I don't think issue would be relevant now since  most of the packages also are supporting Flutter v3 over v2
